torch/jit/frontend.py

```diff
@@ -336,6 +354,8 @@ def build_param(ctx, py_arg, self_name, kwarg_only):
     r = ctx.make_range(py_arg.lineno, py_arg.col_offset, py_arg.col_offset + len(name))
     if getattr(py_arg, 'annotation', None) is not None:
         annotation_expr = build_expr(ctx, py_arg.annotation)
+    elif getattr(py_arg, 'annotation', None) is None and bool(_args_and_types):  # type: ignore[name-defined]
```

Comment:
I think we don't need to check if `py_arg.annotation` is `None` because line 355 already checks that

Reply:
Also, it is more Pythonic to check `none`ness of _args_and_types rather than giving it a default value of `False` and checking boolean value

Reply:
Yeah, I think the check  `py_arg.annotation` is `None` is redundant. I initially thought we would need both, now we can just check for `_args_and_types` `none`ness